### PR TITLE
W-14137568-set-payload-cv

### PR DESCRIPTION
**Pull request by Cristian-Venticinque**

### CODE REVIEW

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+//end::component-title[]
+//
+
+//
+//tag::component-short-description[]
+//     Short description of the form "Do something..." 
+//     Example: "Configure log messages anywhere in a flow."
+
+Updates the payload of the message.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc]: Is it always updating? Maybe "Configures the payload of" ?

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+| Editable name for the component.
+
+| `doc:id` 
+| Automatically generated identifier for the component.
+
+| `value` 
+| Accepts a literal string or DataWeave expression that defines how to set the payload. For example, ``"some string"`` or `#[now()]`. This attribute is required. 
+
+| `mimeType` 
+| The mime type of the value assigned to the payload. For example, `text/plain` or `application/json`.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc]:
```suggestion
| The MIME type of the value assigned to the payload. For example, `text/plain` or `application/json`.
```

**Review by dukesphere — COMMENTED** [review]
some comments

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+| Description
+
+| `doc:name` 
+| Editable name for the component.
+
+| `doc:id` 
+| Automatically generated identifier for the component.
+
+| `value` 
+| Accepts a literal string or DataWeave expression that defines how to set the payload. For example, ``"some string"`` or `#[now()]`. This attribute is required.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc:86]:
```suggestion
| `value` 
| Accepts a literal string or DataWeave expression that defines how to set the payload. For example, ``"some string"`` or `#[now()]`. This attribute is required. 

| `resource`
|  Points to a `.dwl` file containing a DataWeave transformation script. These values are resolved statically (not dynamically). For example, using `<ee:set-payload resource="myscript-${env}.dwl"/>` generates an error.
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc (hunk 1 above)
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc]:
```suggestion
Sets or updates the payload of the message.
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+Updates the payload of the message.
+
+//end::component-short-description[]
+//
+
+//
+//tag::component-long-description[]
+
+The payload can be a literal string or a DataWeave expression. Use the `set-payload` component for simple expressions or transformation.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc]:
```suggestion
The payload can be a literal string or a DataWeave expression. Use the `set-payload` component only for simple expressions or transformations.
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+Updates the payload of the message.
+
+//end::component-short-description[]
+//
+
+//
+//tag::component-long-description[]
+
+The payload can be a literal string or a DataWeave expression. Use the `set-payload` component for simple expressions or transformation. 
+//For complex scenarios, use the xref:acb-component-transform.adoc[Transform Message Component].
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc]:
```suggestion
For complex scenarios, use the xref:acb-component-transform.adoc[Transform Message Component].
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+| The mime type of the value assigned to the payload. For example, `text/plain` or `application/json`.
+
+| `encoding`
+| The encoding of the value assigned to the payload. For example, `UTF-8`.
+
+|===
+
+[WARNING]
+--
+The `mimeType` and `encoding` attributes do not affect a DataWeave expression used as `value`. They only affect the output Mule Message. If a transformation is required, the DataWeave expression must contain an explicit output directive.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc]:
```suggestion
The `mimeType` and `encoding` attributes do not affect a DataWeave expression used as a `value`. These attributes affect the _output_ Mule Message only. If a transformation is required, the DataWeave expression must contain an explicit `output` directive.
```

**Pull request merged**